pkg/testing/fixture_install.go

```diff
@@ -128,23 +133,28 @@ func (f *Fixture) Install(ctx context.Context, installOpts *InstallOpts, opts ..
 
 		// environment variable AGENT_KEEP_INSTALLED=true will skip the uninstall
 		// useful to debug the issue with the Elastic Agent
-		if keepInstalled() {
+		if keepInstalled() && f.t.Failed() {
 			f.t.Logf("skipping uninstall; AGENT_KEEP_INSTALLED=true")
+			return
 		} else {
```

Comment:
Since you have a `return` in the `if` now, I think you can remove this `else`.

Reply:
that's why it's wrong ahaha I fixed on https://github.com/elastic/elastic-agent/pull/3380
the log line cannot always be printed, it needs another check